src/profession.h

```diff
@@ -117,6 +119,9 @@ class profession
         std::vector<mtype_id> pets() const;
         std::vector<bionic_id> CBMs() const;
         std::vector<proficiency_id> proficiencies() const;
+        std::vector<matype_id> ma_known() const;
+        std::vector<matype_id> ma_choices() const;
```

Comment:
I know you're just following the pattern of the rest, but why do these return copies? They could just as easily return `const` references.

Reply:
Good question. There are bunch of methods like that. Maybe because they are vectors?

Reply:
It's the right way to do it, https://stackoverflow.com/questions/15704565/efficient-way-to-return-a-stdvector-in-c, if you need to do something to populate a vector.
They're read-only in your code, resulting in copies unecessarily (it can't just move the vector from the class into the vector wherever you're calling it). It's not like this is a hot code path, so I suppose the efficiency if mutation is needed is potentially worth it?